### PR TITLE
Add [EnforceRange] to all integers

### PR DESCRIPTION
**Pull request by kainino0x**
[`[EnforceRange]`](https://heycam.github.io/webidl/#EnforceRange) is the most-strict and least-surprising mode of `number`-to-integer conversion in WebIDL. (The default is modular, the other option is `[Clamp]`.) `[EnforceRange]` rounds the (double-precision floating point) `number` to an integer, then throws if it's outside of the range of the integer type. It has the benefit that removing this attribute would be a non-breaking change (in that it would change from throwing to not-throwing).

I've been meaning to do this for a long time, but I didn't realize that it was an attribute on the type and not the fields/parameters themselves, or I might have done this before removing all the typedefs. 🙃

I'm not attached to using typedefs, it just makes it slightly more apparent if something diverges from this (IMO) (the word `long` will appear). This PR does not leave any unenforced integer types.

Austin did some quick measurements in Chrome's bindings, and this doesn't seem to impact performance in the non-failure case, for us.

***
<a href="https://pr-preview.s3.amazonaws.com/kainino0x/gpuweb/pull/498.html" title="Last updated on Feb 11, 2020, 3:07 AM UTC (75f212b)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/gpuweb/gpuweb/498/a8d1858...kainino0x:75f212b.html" title="Last updated on Feb 11, 2020, 3:07 AM UTC (75f212b)">Diff</a>

### CODE REVIEW

**Comment by JusSn** [comment]
This change, I like it. Another!

File: spec/index.bs [hunk 1]
```diff
@@ -158,7 +158,10 @@ Type Definitions {#type-definitions}
 ============================
 
 <script type=idl>
-typedef unsigned long long GPUBufferSize;
+typedef [EnforceRange] unsigned long long GPUBufferSize;
+typedef [EnforceRange] unsigned long long enforced_u64;
```
Comment (kvark) [inline spec/index.bs]: shouldn't this be camel case, like `EnforcedU32`?

**Comment by kainino0x** [comment]
I'm out, so please feel free to bikeshed this while I'm out. :)

**Comment by Kangz** [comment]
Discussed and approved the semantics in the [2019-11-25 meeting](https://docs.google.com/document/d/1GOoIl_ZQk6DoS30n7e34R8_YqrokBAKdxGJRy3DDmfo). Needs bikeshedding on the name.

File: spec/index.bs [hunk 2]
```diff
@@ -158,7 +158,17 @@ Type Definitions {#type-definitions}
...
+typedef [EnforceRange] unsigned long GPUBufferDynamicOffset;
+typedef [EnforceRange] unsigned long long GPUFenceValue;
+typedef [EnforceRange] unsigned long GPUStencilValue;
+typedef [EnforceRange] unsigned long GPUSampleMask;
+typedef [EnforceRange] long GPUDepthBias;
+
+typedef [EnforceRange] unsigned long GPUIntegerCoordinate;
+typedef [EnforceRange] unsigned long GPUIndex32;
+typedef [EnforceRange] unsigned long GPUSize32;
+typedef [EnforceRange] long GPUSignedOffset32;
```
Comment (kainino0x) [inline spec/index.bs]: Note: I tried not to go too overboard with the number of different types, so consolidated some of them into these generic ones.

**Comment by kainino0x** [comment]
> Is there even a use for non-enforced vars?

IMO, no.

**Comment by Kangz** [comment]
This was discussed in the [2020-01-06 meeting](https://docs.google.com/document/d/1nESeFHsqOgBxG-vM19THWGf_C6G6fVrxj16PXuaE47Q)

**Comment by kainino0x** [comment]
I'd like to get some resolution from @kvark @jdashg so we can land this.

**Comment by jdashg** [comment]
Approved, but has rebase conflicts. @kainino0x

**Comment by kainino0x** [comment]
Merging after merging with master, and with minor nonfunctional changes (see the last commit)

**Pull request merged**